optaplanner-examples/src/main/java/org/optaplanner/examples/common/experimental/ExperimentalConstraintCollectors.java

```diff
@@ -162,15 +162,17 @@
      * @param startMap Maps the fact to its start
      * @param endMap Maps the fact to its end
      * @param <A> type of the first mapped fact
-     * @param <C> type of the fact endpoints
+     * @param <PointType_> type of the fact endpoints
      * @return never null
      */
-    public static <A, C extends Comparable<C>> UniConstraintCollector<A, IntervalTree<A, C>, ConsecutiveIntervalDataImpl<A, C>>
-            consecutiveIntervals(Function<A, C> startMap, Function<A, C> endMap) {
+    public static <A, PointType_ extends Comparable<PointType_>, DifferenceType_ extends Comparable<DifferenceType_>>
+            UniConstraintCollector<A, IntervalTree<A, PointType_, DifferenceType_>, ConsecutiveIntervalInfo<A, PointType_, DifferenceType_>>
+            consecutiveIntervals(Function<A, PointType_> startMap, Function<A, PointType_> endMap,
+                    BiFunction<PointType_, PointType_, DifferenceType_> differenceFunction) {
```

Comment:
I understand why the `differenceFunction` exists. That said, I'm thinking that perhaps we can have an overload where `PointType_` is guaranteed to be a number, and therefore the difference function is not necessary?

Reply:
I thinking of two overloads: one for numbers, one for temporal (localdatetime etc.).